src/scala/microsoft-spark-3.0.x/src/main/scala/org/apache/spark/sql/api/dotnet/SQLUtils.scala

```diff
@@ -37,4 +43,13 @@ object SQLUtils {
       broadcastVars,
       accumulator)
   }
+
+  def createDataFrameHelper(
+      spark: SparkSession,
+      data: Array[Row],
+      schema: StructType): DataFrame = {
+    val dataAsJavaList = data.toList.asJava
```

Comment:
How about we just do this in `readRowArr` similar to `readMap` so that we don't have have to go thru ` createDataFrameHelper`? We can always come back to support `Array[Row]` if there are future APIs that require it.

Reply:
That was the prior implementation. I think @suhsteve  wanted to keep the design for readList unchanged and hence he suggested this change. I can revert it back if that is what is preferred.